### PR TITLE
Implemented a password policy and added Create User error checks

### PR DESCRIPTION
**Pull request by akkowal2**
- Test for missing values such as username, password, email address, etc
- Validate that the password matches a password policy
- Validate that the username provided does not exceed the max number of characters specified in model
- Validate the email address is a valid address

### CODE REVIEW

File: Server/Communication/testhelper.py [hunk 1]
```diff
@@ -61,6 +62,24 @@ def receive(self, channel_num=1):
...
+			req_enabled, req_list = requirements[req]
+			if req_enabled:
+				password += req_list[0]
+
+		for _ in range(len(password), config.PASSWORD_POLICY['Min Length']):
+			password += 'a'
+
+		return password
+
+
```
Comment (DaBernMon) [inline Server/Communication/testhelper.py]: Two empty lines?

Comment (akkowal2) [reply Server/Communication/testhelper.py]: Maybe I missed that.  I typically like just 1 separating line.

File: Server/User/routehelper.py [hunk 2]
```diff
@@ -111,10 +124,10 @@ def createUser(data):
 				error = "That email is already in use."
 			else:
 				error = str(e)
+		else:
+			error = str(e)
```
Comment (DaBernMon) [inline Server/User/routehelper.py]: This looks weird.. so maybe just ignore, but what about putting error = str(e) at the top and removing the else's?  If an if is caught, overwrite error

Comment (akkowal2) [reply Server/User/routehelper.py]: Sounds good to me.  I'll change that.

File: Server/User/tests.py [hunk 3]
```diff
@@ -12,44 +13,142 @@ def setUp(self):
...
+		self.assertFalse(result["Success"])
+		self.assertEquals(result["Error"], "The password does not meet the password requirements: needs to contain at least 1 lowercase character.")
+
+		config.PASSWORD_POLICY['Requirements']['Lowercase'][0] = False
+		config.PASSWORD_POLICY['Requirements']['Uppercase'][0] = True
+		config.PASSWORD_POLICY['Requirements']['Number'][0] = False
+		config.PASSWORD_POLICY['Requirements']['Symbol'][0] = False
+		result = self.channel.createTestUser({"username":"noUpper", "password":"abcdefg", "email":"t@t.com"})
+		self.assertFalse(result["Success"])
+		self.assertEquals(result["Error"], "The password does not meet the password requirements: needs to contain at least 1 uppercase character.")
```
Comment (DaBernMon) [inline Server/User/tests.py]: Does it still?

File: Server/User/tests.py [hunk 4]
```diff
@@ -12,44 +13,142 @@ def setUp(self):
...
+
+		config.PASSWORD_POLICY['Requirements']['Lowercase'][0] = False
+		config.PASSWORD_POLICY['Requirements']['Uppercase'][0] = False
+		config.PASSWORD_POLICY['Requirements']['Number'][0] = True
+		config.PASSWORD_POLICY['Requirements']['Symbol'][0] = False
+		result = self.channel.createTestUser({"username":"noNumber", "password":"ABCDEFG", "email":"t@t.com"})
+		self.assertFalse(result["Success"])
+		self.assertEquals(result["Error"], "The password does not meet the password requirements: needs to contain at least 1 number character.")
```
Comment (DaBernMon) [inline Server/User/tests.py]: Does it still?

File: Server/User/routehelper.py [hunk 5]
```diff
@@ -112,9 +126,7 @@ def createUser(data):
 			else:
```
Comment (DaBernMon) [inline Server/User/routehelper.py]: No longer needed either then.

**Review by DaBernMon — APPROVED** [review]

**Pull request merged**